src/app/content/sagas/contribution.ts

```diff
@@ -1,47 +1,44 @@
-import { put, takeLatest, call } from 'redux-saga/effects';
+import { put, race, take, takeLatest, call } from 'redux-saga/effects';
 import { go, replace } from 'connected-react-router';
 import { reset } from 'redux-form';
 import {
-  contributionSubmissionFailed,
-  contributionSubmitted,
+  CONTRIBUTION_SUBMISSION_FAILED,
+  CONTRIBUTION_SUBMITTED,
   SUBMIT_CONTRIBUTION,
   SubmitContributionAction
 } from 'app/actions/contribution';
 import { createSubmissionError } from 'app/utils/form';
-import sendEmail from 'api/sendInBlue/sendEmail';
-import createContributionEmail from 'app/background/services/createContributionEmail';
 import { history } from '../store';
-import { captureException } from '../../utils/sentry';
-import { createCallAndRetry } from '../../sagas/effects/callAndRetry';
-
-const sendEmailAndRetry = createCallAndRetry({
-  maximumAttempts: 15, // ~ 1 min 45s
-  onFinalError: () => {
-    captureException(new Error('Could not send contribution email'));
-  }
-});
 
 export function* submitContributionSaga({
-  payload: contribution,
   meta: { form, resolve, reject }
 }: SubmitContributionAction) {
   try {
-    yield sendEmailAndRetry(sendEmail, createContributionEmail(contribution));
-
-    yield put(contributionSubmitted(contribution));
+    const { contributionSubmittedAction } = yield race({
+      contributionSubmittedAction: take(CONTRIBUTION_SUBMITTED),
+      contributionSubmissionFailedAction: take(CONTRIBUTION_SUBMISSION_FAILED)
+    });
 
     if (form) {
-      // reset history, form and redirect to success page
-      yield put(go(-history.entries.length));
-      yield put(replace('/contribute/submitted'));
-
-      yield put(reset(form));
-
-      yield call(resolve);
+      if (contributionSubmittedAction) {
+        // reset history, form and redirect to success page
+        yield put(go(-history.entries.length));
+        yield put(replace('/contribute/submitted'));
+
+        yield put(reset(form));
+
+        yield call(resolve);
+      } else {
+        // @todo we should be able to extract the error from `contributionSubmissionFailedAction`
```

Comment:
Maybe `JSON.stringify` it before sending it to tab ?

Reply:
Good suggestion, I'll try to modify the *"action socket"* code and see where it's going.

Reply:
#816